plugin/pkg/scheduler/factory/factory.go

```diff
@@ -237,6 +240,29 @@ func (f *ConfigFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 	}, nil
 }
 
+func (f *ConfigFactory) getFitSchedulerPod() *api.Pod {
+	pod := f.PodQueue.Pop().(*api.Pod)
+	if f.podFitScheduler(pod) {
+		glog.V(2).Infof("About to try and scheduler pod %v", pod.Name)
+		return pod
+	}
+	return f.getFitSchedulerPod()
```

Comment:
I'm not sure recursion is the best approach here. I would suggest something like
```
for {
   pod := f.PodQueue.Pop().(*api.Pod)
   if f.responsibleForPod(pod) {
      return pod
   }
}
```

Since Pop() blocks when the queue is empty, I think this works.
https://github.com/kubernetes/kubernetes/blob/d42030170b2b20cb61cb79dff7d095998b7ee7b1/pkg/client/cache/fifo.go#L27